temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -2049,48 +2085,62 @@ private void timeoutActivity(long scheduledEventId, TimeoutType timeoutType, int
   }
 
   @Override
-  public void startNexusTask(long scheduledEventId, RespondNexusTaskCompletedRequest request) {
+  public void startNexusOperation(
+      long scheduledEventId, String clientIdentity, StartOperationResponse.Async resp) {
     update(
         ctx -> {
           StateMachine<NexusOperationData> operation = getPendingNexusOperation(scheduledEventId);
-          operation.action(StateMachines.Action.START, ctx, request, 0);
-          operation.getData().identity = request.getIdentity();
+          operation.action(StateMachines.Action.START, ctx, resp, 0);
+          operation.getData().identity = clientIdentity;
+          scheduleWorkflowTask(ctx);
         });
   }
 
   @Override
-  public void completeNexusTask(long scheduledEventId, RespondNexusTaskCompletedRequest request) {
+  public void cancelNexusOperation(NexusOperationRef ref) {
     update(
         ctx -> {
-          StateMachine<NexusOperationData> operation = getPendingNexusOperation(scheduledEventId);
-          throwIfOperationTokenDoesntMatch(request.getTaskToken(), operation.getData());
-          if (request.getResponse().hasCancelOperation()) {
-            operation.action(Action.CANCEL, ctx, request, 0);
-          } else {
-            operation.action(StateMachines.Action.COMPLETE, ctx, request, 0);
-          }
-          // nexusOperations.remove(scheduledEventId); // TODO(pj): server currently does not delete
+          StateMachine<NexusOperationData> operation =
+              getPendingNexusOperation(ref.getScheduledEventId());
+          throwIfOperationRefDoesntMatch(ref.toBytes(), operation.getData());
+          throwIfOperationNotInFlight(operation.getState());
```

Comment:
You should ignore this instead of throwing (unless you're catching this somewhere else) but this is expected due to ordering issues and timeouts.

(Apply this here and everywhere else where you put these assertions)